src/Sylius/Bundle/AdminBundle/Notification/GusNotificationProvider.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\AdminBundle\Notification;
+
+use Psr\Clock\ClockInterface;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Sylius\Bundle\CoreBundle\SyliusCoreBundle;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Contracts\Cache\CacheInterface;
+use Symfony\Contracts\Cache\ItemInterface;
+
+final class GusNotificationProvider implements NotificationProviderInterface
+{
+    public const LATEST_SYLIUS_VERSION_CACHE_KEY = 'latest_sylius_version';
+
+    public function __construct(
+        private readonly ClientInterface $client,
```

Comment:
Is it possible to define whole class as `readonly`, so it won't be necessary to specify over here?